src/_guides/language/language-tour.md

```diff
@@ -3862,9 +3863,30 @@ print(names is List<String>); // true
 ### Restricting the parameterized type
 
 When implementing a generic type,
-you might want to limit the types of its parameters.
+you might want to limit the types of its parameters
+to be a subtype of a particular type.
 You can do this using `extends`.
 
+A common use case is ensuring that a type is non-nullable
+by making it a subtype of `Object`
+(instead of the default, [`Object?`][top-and-bottom]).
+
+<?code-excerpt "misc/lib/language_tour/generics/misc.dart (non-nullable)"?>
+```dart
+class Foo<T extends Object> {
+  // Foo works on objects with the non-nullable type T.
+}
+```
+
+How does `Foo<T extends Object>` differ from `Foo<Object>`?
```

Comment:
This sentence doesn't really make sense.  It's mixing categories.  Do you mean `Foo<T>` for the latter?  I actually have no idea what this sentence is trying to say, to be honest.

Reply:
This is an attempt to fix #2659.

Reply:
I still don't know what this is trying to say.  Let me try to expand out a bit.  The syntax `Foo<T extends Object>` only makes sense as part of a class declaration: `class Foo<T extends Object>`.  That is, I can use it as a type: `int f(Foo<T extends Object> parameter) => 3)` is not valid syntax.  The syntax `Foo<Object>` on the other hand, is *technically* valid in both a declaration and a use, but almost certainly is not what you want in a declaration.  Writing `class Foo<Object> {}` is exactly the same as writing `class Foo<T> {}`.  That is, `Object` here does *not* refer to the core `Object` type, but is just a bound type variable, just like `T`.  The syntax `int f(Foo<Object> parameter) => 3` on the other hand means "f is a function which takes a parameter of type `Foo<Object>`.  That is, `Object` is the type *argument* passed to `Foo`.  

So my confusion is that this sentence, I think, is mixing declarations and uses.  The discussion in the #2659, after the initial comment (which seemed to share the same confusion), became more about the difference between subtype and generic polymorphism.  If the goal here is to elaborate on that, I think a different structure is warranted.

Reply:
OK, I've removed that note.